### DISCUSSION TITLE
consult-ripgrep with git submodules

### DISCUSSION THREAD

**Discussion by nicolas-graves** [minad/consult#1097] [Ideas]
Hi @minad ,

I have a project where I use git submodules. When I consult-ripgrep in a submodule, I expect to ripgrep in this git submodule rather than in the whole project. There seems to be a variable to govern that in `project.el`: `project-vc-merge-submodules`. It would be nice if some `consult` commands could honor that variable by executing in the subproject rather than in the project as a whole. 

This way we could properly separate project and subprojects : with a `.rgignore` that ignores submodules, I would be able to ripgrep only where I want both in the project and in each subproject. 

WDYT?

**Comment by minad** [comment]
Right now Consult can only query for the current project directory and doesn't have any other further more fine-grained project information. Therefore this seems mostly out of scope for Consult.

**Reply by nicolas-graves** [reply]
OK, thanks for your quick reply!

**Reply by minad** [reply]
You're welcome. As a workaround it should be rather easy to replace `consult-project-function` with a custom function, which returns the submodule directory only. I wonder why `project-current` doesn't already do that and return the submodule project only if the variable is set to merge=nil?

**Reply by nicolas-graves** [reply]
I'll take a look soon and comment here once I've found a suitable solution. Thanks for your work.

**Reply by nicolas-graves** [reply]
Actually `project-vc-merge-submodules` is currently only used when listing buffers or files, it has no influence on `project-current`.

**Reply by nicolas-graves** [reply]
After further research, `project-current` can't handle this because it calls `project--find-in-directory` which then calls `project-try-vc` through `project-find-functions`. 
`project-try-vc` has an unclear FIXME that seems to be what is missing to make that work.

**Reply by nicolas-graves** [reply]
The `FIXME` is not enough, it seems that even when clearing the cache, the following code is setting the cache back to the parent, it seems written this way : 
``` elisp
        (when (and
               (eq backend 'Git)
               (project--vc-merge-submodules-p root)
               (project--submodule-p root))
          (let* ((parent (file-name-directory (directory-file-name root))))
            (setq root (vc-call-backend 'Git 'root parent))))

```

**Reply by minad** [reply]
Any updates on this? Can project.el be fixed accordingly?
